### ISSUE TITLE
Sending home assistant and standard mqtt messages together

### ISSUE THREAD

**Issue by M0ebiu5** [bachya/ecowitt2mqtt#935]
It would be nice, if one could enable home assistant and standard protocoll for the same unit.

Use case is, if you want to migrate from one to another, it's much easier not to have to switch between protocolls...

**Comment by bachya** [comment]
(Related to https://github.com/bachya/ecowitt2mqtt/issues/640 in that it's about another publisher type.)

I'm not familiar with the Homematic protocol. Can you share more information or documentation?

**Comment by M0ebiu5** [comment]
sorry - it should be home assistant...

**Comment by bachya** [comment]
This was added in [2023.06.0](https://github.com/bachya/ecowitt2mqtt/releases/tag/2023.06.0) (via https://github.com/bachya/ecowitt2mqtt/pull/606). Are you experiencing a bug?

**Comment by M0ebiu5** [comment]
If i switch to home assistant, i dont get the "normal" mqtt messages...i cant enable both. I've tried setting up 2 different sections in the config file, but then i get duplicate key...if this constraint would be removed, mabe it would work?

**Comment by bachya** [comment]
I would need to see your config file to comment further there. I just spun up a 2024.01.2 instance and everything worked as expected:

```
ecowitt2mqtt --mqtt-broker 127.0.0.1 --mqtt-user ecowitt --mqtt-password password --mqtt-topic Test --hass-discovery
2024-04-01 11:36:43,000 | INFO | Starting ecowitt2mqtt (version 2024.01.2)
2024-04-01 11:36:46,361 | INFO | Published to Test
2024-04-01 11:36:46,363 | INFO | Published to Home Assistant MQTT Discovery
```

Notice how I'm providing both `--mqtt-topic` and `--hass-discovery` flags. With this configuration, I see all expected topics:

![CleanShot 2024-04-01 at 11 36 22](https://github.com/bachya/ecowitt2mqtt/assets/47216/c9f40b52-08e4-4c6b-904d-b81474621a00)

**Comment by M0ebiu5** [comment]
This is the command line i had to run it as a service. I also provided mqtt-topic and hass-discovery. You can see in the output, that it only sends hass messages.

` ecowitt2mqtt --mqtt-broker=big --mqtt-username=*** --mqtt-password="***" --mqtt-topic=eco1 --input-data-format=ecowitt --port 8282 --output-unit-system metric --hass-discovery

2024-04-02 13:14:40,666 | INFO | Starting ecowitt2mqtt (version 2023.05.0)
2024-04-02 13:16:06,393 | INFO | Published to Home Assistant MQTT Discovery
2024-04-02 13:18:14,200 | INFO | Published to Home Assistant MQTT Discovery`

**Comment by bachya** [comment]
> This is the command line i had to run it as a service. I also provided mqtt-topic and hass-discovery. You can see in the output, that it only sends hass messages.
> 
> 
> 
> ```
> ecowitt2mqtt --mqtt-broker=big --mqtt-username=*** --mqtt-password="***" --mqtt-topic=eco1 --input-data-format=ecowitt --port 8282 --output-unit-system metric --hass-discovery
> 
> 
> 
> 2024-04-02 13:14:40,666 | INFO | Starting ecowitt2mqtt (version 2023.05.0)
> 
> 2024-04-02 13:16:06,393 | INFO | Published to Home Assistant MQTT Discovery
> 
> 2024-04-02 13:18:14,200 | INFO | Published to Home Assistant MQTT Discovery
> ```

Your logs show that you are running 2023.05.0. As I mentioned [above](#issuecomment-2030154701), the functionality you're looking for was released in 2023.06.0.

**Comment by M0ebiu5** [comment]
I just saw that i was running an older version...sorry!

Thanks for your support and your efforts with ecowitt2mqtt!
